dmoj/cptbox/utils.py

```diff
@@ -1,11 +1,62 @@
+import errno
 import io
+import mmap
+import os
+from tempfile import NamedTemporaryFile
+from typing import Optional
 
 from dmoj.cptbox._cptbox import memory_fd_create, memory_fd_seal
+from dmoj.cptbox.tracer import FREEBSD
 
 
 class MemoryIO(io.FileIO):
```

Comment:
Where/how do we end up deleting the memory backing these objects once they're no longer necessary to grade a case?

Reply:
It's closed when `MemoryIO.close` is called or it's garbage collected, since `FileIO` constructor takes ownership of the file descriptor unless `closefd=False` is passed.